File: bottles.jl
#!/usr/bin/env julia

#=
"99 Bottles of Beer" in Julia

by Matthew Morris <user="anglus", domain="gmail.com">
August 3, 2016 (tested on Julia 0.4.6).

Fairly straighforward version using a function and the ternary operator.

This software is in the public domain.
=#

function bottles(n::Int)
	text = n == 0 ? "no more" : string(n)
	text = string(text, n == 1 ? " bottle" : " bottles")
	text = string(text, " of beer")
	text
end

for num = 99:-1:0
	println(ucfirst(bottles(num)), " on the wall, ", bottles(num), ".")
	print(num == 0 ? "Go to the store and buy some more, " : "Take one down and pass it around, ")
	println(bottles(num > 0 ? num - 1 : 99), " on the wall.")
	println()
end

File: bottles.nim
import strutils

proc bottles(n: int): string =
    var text: string 
    text = if n == 0: "no more" else: intToStr(n)
    text &= (if n == 1: " bottle" else: " bottles")
    text &= " of beer"
    return text

for num in countdown(99, 0):
    echo bottles(num).capitalize & " on the wall, " & bottles(num) & "."
    write(stdout, if num == 0: "Go to the store and buy some more, " else: "Take one down and pass it around, ")
    echo bottles(if num > 0: num - 1 else: 99) & " on the wall."
    echo() 
